### ISSUE TITLE
Installation Navison 2016 and 2017 fail due to authorisation issue

### ISSUE THREAD

**Issue by mheijkant** [gunnargestsson/AdvaniaGIT#2]
I’m stuck on both the Nav 2016 and Nav 2017 installation.
Because most of the posted You-Tube demo’s from yur "Installing AdvaniaGIT – Video Help" blog don’t contain sound it’s pretty difficult to follow. 
For instance why and how to modify the GITSettings.json and BranchSettings.json. 

On the Nav 2016 Installation I receive following error while running the “Update-NAVSource.ps1” file:

Downloading CU information from Microsoft Blog...
Downloading installation for 2016 W1...
CU 22 NAV 2016 W1.zip already downloaded...
Extracting C:\AdvaniaGIT\Download\CU 22 NAV 2016 W1.zip to C:\AdvaniaGIT\2016W1...
Copying backup file from installation media...
Requesting new NAV Environment for branch master
System.Management.Automation.MethodInvocationException Exception calling "GetResponse" with "0" argument(s): "De externe server heeft een fout geretourneerd: (530) Niet aangemeld."
Press enter to continue...:

On the Nav 2017 installation I receive following error while running the “Update-NAVSource.ps1” file:

Copying backup file from installation media...
Set-Content : Could not open the alternate data stream '\AdvaniaGIT\Data\BranchSettings.Json' of the file 'C:\AdvaniaGI
T\C'.
At C:\AdvaniaGIT\Scripts\AdvaniaGIT\Get-BranchSettings.ps1:24 char:9
+  Set-Content -Path (Join-Path (Split-Path -Parent (Split-Path  ...
+ CategoryInfo          : ObjectNotFound: (C:\AdvaniaGIT\C:String) [Set-Content], FileNotFoundException
+ FullyQualifiedErrorId : GetContentWriterFileNotFoundError,Microsoft.PowerShell.Commands.SetContentCommand

Requesting new NAV Environment for branch master
System.Management.Automation.MethodInvocationException Exception calling "GetResponse" with "0" argument(s): "De externe server heeft een fout geretourneerd: (530) Niet aangemeld."
Press enter to continue...: .

**Comment by gunnargestsson** [comment]
First try to update everything from GitHub.  I am working on a lot up updated.

Can you please try to debug the functionality ?
You start Powershell ISE in Administration Mode.
Open C:\AdvaniaGIT\TestDevel.ps1

Get-Module -Name AdvaniaGIT | Remove-Module
$ScriptsPath = Join-Path $PSScriptRoot 'Scripts' 
. (Join-Path $ScriptsPath 'Start-CustomAction.ps1') "<Your GIT Folder Here>" "Update-NAVSource.ps1" $false

You can set break points and step into and over until you hit the error.

**Comment by gunnargestsson** [comment]
Also make sure, if you are using an ftp server to have the correct folder structure

https://github.com/gunnargestsson/AdvaniaGIT/wiki/FTP-Server

**Comment by mheijkant** [comment]
At this moment, after  the latest update, I now receive following error when running “Update GIT from latest CU”, but the installation continues.

Combine-Settings : The term 'Combine-Settings' is not recognized as the name of a cmdlet, function, script file, or ope
rable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again
.
At C:\Users\michel\Documents\AdvaniaGIT\SetupLocalCopy.ps1:63 char:20
+     $GITSettings = Combine-Settings $GITSettings $NewGITSettings
+                    ~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Combine-Settings:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Trying to get the debug script to run with I made following adjustment, to make it work for me.

C:\AdvaniaGIT\Scripts\Start-CustomAction.ps1 "C:\Users\michel\Documents\nav2016" "Update-NAVSource.ps1" $false
Where "C:\Users\michel\Documents\nav2016""is a random directory becaus an empty "" didn't work. I'm not a ProwerShell-Pro, so please forgive me if it's not up to standard.

When Debugging the script, it failed on making an FTP connection.
I suppose I have 2 options here:
1.	Not using FTP, which also helped to overcome the first mentioned 530 error.
2.	proving the correct FTP setting. But I must mention here that I have to create a 2016 or/and 2017 directory manually for both nav 2016 and nav2017 installations.

So it reads ftp://myserver:22/2016 and ftp://myserver:22/2017. Also notice the extra port that I had to provide because of our own FTP installation.

Next error I receive now is:

Requesting new NAV Environment for branch master
Restoring database...
System.Management.Automation.CommandNotFoundException The term 'New-NAVDatabase' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
Press enter to continue...:

This is caused in Script Restore-NAVBackup.ps1 on line 30.
I don't recall seeing that a Navision Module is imported, so this command is not recognised.
Can you help me further with this next issue.

**Comment by gunnargestsson** [comment]
Just updated the setup script to the repository.  Update and rerun the SetupLocalCopy.ps1.

**Comment by mheijkant** [comment]
The mentioned issue when running the SetupLocalCopy.ps1 is resolved. 
I also reported the " 'New-NAVDatabase' is not recognized " issue.
I didn't mention another issue on running SetupLocalCopy.ps1, but maybe it's related to this one.
Also here its stated that a navisoin command is not recognised.

When choosing the option "Perform NAV Environment Configuration (required after NAV installation) ? (Default = Yes):" and answer Yes, I see following error and others:
Stopping Services for version 10
Get-NAVServerInstance : The term 'Get-NAVServerInstance' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At C:\AdvaniaGIT\Scripts\Prepare-NAVEnvironment.ps1:18 char:9
+ Get-NAVServerInstance | Where-Object -Property Version -Match ...
+ CategoryInfo          : ObjectNotFound: (Get-NAVServerInstance:String) [], mmandNotFoundException
+ FullyQualifiedErrorId : CommandNotFoundException

**Comment by gunnargestsson** [comment]
Do you have NAV 2017 installed?  Seems like the NAV Admin Tools for 2017 are having troubles loading.

**Comment by mheijkant** [comment]
You where correct about this. I did not have an instance of Version 10 installed. So the error didn't occure after installing a nav 10 instance. But still the script gives errors on checking for version 8 and 9. Mayby you can do a check before trying to stop and restart a nav 8, 9 or 10 instances.

summarizing: 

The initial issue concerning the authorisation was caused by misconfiguring the FTP settings.
the succeeding issue about the navison command not being recognised was caused because I had no initial Navsion Installation. 

Having a few concluding questions about the demo-Installation. 
Maybe its because the video's on this page "http://www.dynamics.is/?p=2830" are without sound, i'm missing some initial information.
1. I expected that you don't need an initial Navsion installation when starting the demo's, but without, the powershell scripts don't recognise the navision command's. So, do you need an initial Nav 2016 or nav 2017 installation before starting the demo's or is there a custom action for this in Sourcetree/AdvaniaGit.
2. Is it possible to catagorize the custom actions in submenus or seperated with splitlines.

**Comment by gunnargestsson** [comment]
Hi
I updated the prepare script to solve the issue with finding installed instances.

1. You need the initial installation
2. I have not found a way to use submenus.  We could request changes to SourceTree

**Comment by mheijkant** [comment]
Re-installed AdvaniaGit and initial Navsion 13682 installation, but now the navison 10 version is not recognised. So the shared port config is no longer running.

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        29-8-2017     12:17              0 StartPowerShell.cmd
Please update GITSettings.Json, DockerSettings.json and NAVVersions.Json to match your environment
Perform Remote Administration Module Installation ? (Default = No):
Perform NAV Environment Configuration (required after NAV installation) ? (Default = Yes):
Press any key to continue...:

**Comment by gunnargestsson** [comment]
Just updated Prepar-NAVEnvironment - try again please

**Comment by mheijkant** [comment]
Thanks, the update solved this minor issue.
Thanks so much, for your responses.
